packages/react-scripts/config/webpack.config.dev.js

```diff
@@ -232,5 +232,11 @@ module.exports = {
     fs: 'empty',
     net: 'empty',
     tls: 'empty'
+  },
+  // Turn off performance hints during development because we don't do any
+  // splitting or minification in interest of speed. These warnings become
+  // cumbersome.
+  performance: {
+    hints: false
```

Comment:
Since this is turned off completely during development and by default set to `warning` in the production config, it might come as a surprise to users to then have to tweak their app for code-splitting, although extracting css should bring the file size down. @TheLarkInn gives a reason to [leave it enabled](/webpack/webpack.js.org/pull/500/files#r92665821) in dev builds. We could tweak the `maxAssetSize` and `maxEntrypointSize` options to reasonable file sizes instead. What do you think?